internal/provider/project_resource.go

```diff
@@ -211,15 +284,29 @@ func (r *projectResource) Update(ctx context.Context, req resource.UpdateRequest
 	updateProjectSettingsRequest := *unleash.NewUpdateProjectEnterpriseSettingsSchemaWithDefaults()
 	updateProjectSettingsRequest.SetMode(mode)
 
-	updateSettingsResponse, err := r.client.ProjectsAPI.UpdateProjectEnterpriseSettings(ctx, *state.Id.ValueStringPointer()).UpdateProjectEnterpriseSettingsSchema(updateProjectSettingsRequest).Execute()
+	if featureNaming, set := expandFeatureNaming(plan.FeatureNaming, &resp.Diagnostics); set {
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		updateProjectSettingsRequest.SetFeatureNaming(*featureNaming)
+	} else if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if linkTemplates, set := expandLinkTemplates(plan.LinkTemplates, &resp.Diagnostics); set {
```

Comment:
Not quite understanding what set is doing for us here. If set is false then this always an error and we could just skip the check on 297 surely?